reqwest-tracing/src/reqwest_otel_span_builder.rs

```diff
@@ -96,10 +97,20 @@ pub struct DefaultSpanBackend;
 
 impl ReqwestOtelSpanBackend for DefaultSpanBackend {
     fn on_request_start(req: &Request, ext: &mut Extensions) -> Span {
-        let name = ext
-            .get::<OtelName>()
-            .map(|on| on.0.as_ref())
-            .unwrap_or("reqwest-http-client");
+        let name = if let Some(name) = ext.get::<OtelName>() {
+            Cow::Borrowed(name.0.as_ref())
+        } else if let Some(path_names) = ext.get::<OtelPathNames>() {
+            path_names
+                .find(req.url().path())
+                .map(|path| Cow::Owned(format!("{} {}", req.method(), path)))
+                .unwrap_or_else(|| {
+                    warn!(target: "No OTEL path name found", path = %req.url().path());
```

Comment:
the whole point is not to leak the path 😆 the existence of a trace should hopefully help pinpoint what request is missing the path name
```suggestion
                    warn!("no OTEL path name found");
```

Reply:
👍 Updated